fabric-component-access-api-v1/src/main/java/net/fabricmc/fabric/api/component/access/v1/ComponentAccess.java

```diff
@@ -0,0 +1,189 @@
+	 */
+	default T get() {
+		return get(null, null);
+	}
+
+	/**
+	 * Retrieves the component with the given access parameters and applies the
+	 * given action if the value is not {@link ComponentType#absent()}.
+	 *
+	 * @param side Side from which the component is being accessed
+	 * @param id Identifier of a specific component or sub-component
+	 * @param action Action to be applied to a non-absent component
+	 * @return {@code true} if a non-absent component was successfully obtained and the action was applied to it
+	 */
+	default boolean acceptIfPresent(/* @Nullable */ Direction side, /* @Nullable */ Identifier id, Consumer<T> action) {
+		final T svc = get(side, id);
+
+		if (svc != componentType().absent()) {
+			action.accept(svc);
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Retrieves the component with the given access parameters and applies the
+	 * given action if the value is not {@link ComponentType#absent()}.
+	 *
+	 * @param side Side from which the component is being accessed
+	 * @param action Action to be applied to a non-absent component
+	 * @return {@code true} if a non-absent component was successfully obtained and the action was applied to it
+	 */
+	default boolean acceptIfPresent(/* @Nullable */ Direction side, Consumer<T> action) {
+		return acceptIfPresent(side, null, action);
+	}
+
+	/**
+	 * Retrieves the component with the given access parameters and applies the
+	 * given action if the value is not {@link ComponentType#absent()}.
+	 *
+	 * @param id Identifier of a specific component or sub-component
+	 * @param action Action to be applied to a non-absent component
+	 * @return {@code true} if a non-absent component was successfully obtained and the action was applied to it
+	 */
+	default boolean acceptIfPresent(/* @Nullable */ Identifier id, Consumer<T> action) {
+		return acceptIfPresent(null, id, action);
+	}
+
+	/**
+	 * Retrieves the component with default access parameters and applies the
+	 * given action if the value is not {@link ComponentType#absent()}.
+	 *
+	 * @param action Action to be applied to a non-absent component
+	 * @return {@code true} if a non-absent component was successfully obtained and the action was applied to it
+	 */
+	default boolean acceptIfPresent(Consumer<T> action) {
+		return acceptIfPresent(null, null, action);
+	}
+
+	/**
+	 * Retrieves the component with the given access parameters and applies the
+	 * given function if the component instance is not {@link ComponentType#absent()}, returning the result.
+	 *
+	 * @param side Side from which the component is being accessed
+	 * @param id Identifier of a specific component or sub-component
+	 * @param function Function to be applied to a non-absent component
+	 * @return Function result if a non-absent component was successfully obtained or {@code null} otherwise
+	 */
+	default <V> V applyIfPresent(/* @Nullable */ Direction side, /* @Nullable */ Identifier id, Function<T, V> function) {
```

Comment:
```suggestion
	 */
	 /* @Nullable */
	default <V> V applyIfPresent(/* @Nullable */ Direction side, /* @Nullable */ Identifier id, Function<T, V> function) {
```
I think documenting the nullability of the return value explicitly should be useful for when we switch to actual annotations (also applies to overloads).

Reply:
I plan to merge the loom update allowing the use of the annotations after 1.16.3 (or before if it doesnt come out in a day or two), so these comments can be replaced once thats done.

Reply:
Added